.pre-commit-config.yaml

```diff
@@ -0,0 +1,39 @@
+default_language_version:
+  python: python3.7
+
+fail_fast: true
+
+repos:
+- repo: https://github.com/pre-commit/pre-commit-hooks
+  rev: v4.2.0
+  hooks:
+    - id: check-ast   # checks Python syntax
+    - id: check-json  # checks JSON syntax
+    - id: check-yaml  # checks YAML syntax
+    - id: check-toml  # checks TOML syntax
+    - id: end-of-file-fixer  # checks there is a newline at the end of the file
+    - id: trailing-whitespace  # trims trailing whitespace
+    - id: check-merge-conflict  # checks for no merge conflict strings
+    - id: check-shebang-scripts-are-executable  # checks all shell scripts have executable permissions
+    - id: mixed-line-ending  # normalizes line endings
+    - id: no-commit-to-branch  # prevents committing to master
+    - id: pretty-format-json  # indents and sorts JSON files
+
+- repo: https://github.com/psf/black
+  rev: 22.6.0  # IMPORTANT: keep this aligned with the black version in setup.cfg
+  hooks:
+  - id: black
+
+# These can fail if some dependencies are missing. Also untested on Windows.
+- repo: local
+  hooks:
+    - id: pydoc-markdown
+      name: Update API documentation (slow)
+      entry: .github/utils/pydoc-markdown.sh
+      language: script
+      types: [bash]
+      pass_filenames: false
+      always_run: true
+
+
+# TODO we can make mypy and pylint run at this stage too, once their execution gets normalized
```

Comment:
what do you mean by "once their execution gets normalized"? `pyliny` is too slow in my opinion to make a nice commit hook. Mypy works fine for DC

Reply:
mypy is in a very bad situation right now for OSS. Just have a look here: https://github.com/deepset-ai/haystack/blob/632cd1c141a8b485c6ef8695685d2d8eef3ca50f/.github/workflows/tests.yml#L50-L61

When this will get fixed it will be reasonable to add it as a hook too